code/modules/mining/machine_redemption.dm

```diff
@@ -205,142 +205,141 @@
 		return
 	interact(user)
 
-/obj/machinery/mineral/ore_redemption/interact(mob/user)
-	var/dat = "This machine only accepts ore. Gibtonite and Slag are not accepted.<br><br>"
-	dat += "Current unclaimed points: [points]<br>"
+/obj/machinery/mineral/ore_redemption/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "ore_redemption_machine", "Ore Redemption Machine", 400, 500, master_ui, state)
+		ui.open()
 
+/obj/machinery/mineral/ore_redemption/ui_data(mob/user)
+	var/list/data = list()
+	data["unclaimedPoints"] = points
 	if(inserted_id)
-		dat += "You have [inserted_id.mining_points] mining points collected. <A href='?src=\ref[src];eject_id=1'>Eject ID.</A><br>"
-		dat += "<A href='?src=\ref[src];claim=1'>Claim points.</A><br><br>"
+		data["hasID"] = TRUE
+		data["claimedPoints"] = inserted_id.mining_points
 	else
-		dat += "No ID inserted.  <A href='?src=\ref[src];insert_id=1'>Insert ID.</A><br><br>"
+		data["hasID"] = FALSE
```

Comment:
This is not needed. Not setting hasID will already return false in the template.